app/src/main/java/ch/epfl/cs311/wanderwave/model/localDb/LocalRecentlyPlayedRepository.kt

```diff
@@ -0,0 +1,46 @@
+package ch.epfl.cs311.wanderwave.model.localDb
+
+import ch.epfl.cs311.wanderwave.model.data.Track
+import ch.epfl.cs311.wanderwave.model.repository.RecentlyPlayedRepository
+import ch.epfl.cs311.wanderwave.model.repository.TrackRepository
+import java.time.Instant
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.launch
+
+class LocalRecentlyPlayedRepository
+@Inject
+constructor(
+    private val recentlyPlayedDao: RecentlyPlayedDao,
+    private val trackRepository: TrackRepository,
+    private val ioDispatcher: CoroutineDispatcher
+) : RecentlyPlayedRepository {
+  override fun addRecentlyPlayed(track: Track, timestamp: Instant) {
```

Comment:
Maybe put a default argument for the timestamp as it's quite rare we will not put now.